### PR TITLE
Move useLoadMore functionality in the Load More component

### LINKED ISSUE #1235: Audio search result page non-interactive if there are fewer than 20 results
## Description

SSR-ed audio search page with a few results is non-interactive 

## Reproduction

1. Go to https://search-staging.openverse.engineering/search/audio/?q=galah 
2. You cannot see the Waveform, the search type switcher button is smaller than it should be, and you clicking anywhere doesn't do anything.

## Additional context

The problem is in the `useLoadMore` composable, and not the waveform as it might seem from the console error messages.

## Resolution

- [x] 🙋 I would be interested in resolving this bug.

### LINKED ISSUE #1301: If there are 0 audio results on the All Content view (but some images), the Load more button is not clickable
## Description

Example: https://search-staging.openverse.engineering/search/?q=%F0%9F%90%88

Go to that search and try to use the load more button, it is unresponsive.

Originally reported by @obulat 

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1235 by @obulat
Fixes #1301 by @sarayourfriend 
Replacement of #1236

## Description

This PR moves all the logic for `VLoadMore` button to the `VLoadMore` component, and heavily uses the data available in the media store.
This solves the problem where because of an error in `VLoadMore`, there was a server-client mismatch, and the whole app was re-rendering, and was not interactive on SSR-ed search results.

This branch is based on `type_base_components` because it uses TS and `VButton`. It is, therefore, blocked by #1233.

## Testing Instructions

Go to localhost:8443/search/audio/?q=galah
You should see a single result, but other than that, the page should be interactive (search type button in the header should work), and the waveform should be displayed.
Compare it with the behavior on main by going to https://search-staging/search/audio/?q=galah

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by sarayourfriend** [comment]
For me when I go to `http://192.168.1.105:8443/search/audio?q=galah` it also shows a "load more" button. When I click it the page switches to the "no results" page :thinking:

**Review by sarayourfriend — COMMENTED** [review]
Also noticed that if you enable a file extension filter that results in 0 results (like mp3 for galah audio search) and then toggle it back off, it still shows no results.

**Comment by obulat** [comment]
The reason for failures described here is the fact that instead of using the store state from the server, we are re-initializing it on the client, for all the Pinia stores.
This means that even though we got results on the server, the result counts on the client are set to 0 in the beginning. FetchState is in the initial state, and `page` is set to undefined, so you can click on the Load More button.

**Review by sarayourfriend — COMMENTED** [review]
This is already an improvement as the page doesn't crash, at least, but there is still something wrong with the load more logic.

When I visit `http://localhost:8443/search/audio?q=galah` on this branch the page doesn't crash and remains interactive. However there is a load more button (even though there's only a single result). If I click it, then a "no results" page will show.

File: src/components/VLoadMore.vue [hunk 1]
```diff
@@ -12,33 +13,43 @@
...
 
-    const onLoadMore = () => {
-      emit('onLoadMore')
+      await mediaStore.fetchMedia({
+        shouldPersistMedia: true,
+      })
     }
+    const isFetching = computed(() => mediaStore.fetchState.isFetching)
+
+    const buttonLabel = computed(() => i18n.t('browse-page.load'))
```
[the remark this answers is not in the document]
Comment (obulat) [reply src/components/VLoadMore.vue:46]: No it doesn't 😆 I think I wrote it that way when I was only starting to learn the way Composition API worked.

**Comment by obulat** [comment]
> This is already an improvement as the page doesn't crash, at least, but there is still something wrong with the load more logic.
> 
> When I visit `http://localhost:8443/search/audio?q=galah` on this branch the page doesn't crash and remains interactive. However there is a load more button (even though there's only a single result). If I click it, then a "no results" page will show.

Sorry for not stating it clearly, but this PR should only work after the media PR is merged.

**Review by sarayourfriend — APPROVED** [review]
LGTM! Massive improvement :tada:

File: src/components/VLoadMore.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (krysal) [reply src/components/VLoadMore.vue:46]: +1 to putting this directly on the template 🙂

**Pull request merged**